pkg/adapters/kuadrant/client.go

```diff
@@ -0,0 +1,44 @@
+package kuadrant
+
+import (
+	"context"
+	"fmt"
+	kctlrv1beta1 "github.com/kuadrant/kuadrant-controller/apis/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilRuntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientGoScheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+)
+
+var (
+	Client        client.Client
+	RuntimeScheme = runtime.NewScheme()
+)
+
+func init() {
+	utilRuntime.Must(clientGoScheme.AddToScheme(RuntimeScheme))
+	utilRuntime.Must(kctlrv1beta1.AddToScheme(RuntimeScheme))
+
+	configuration := config.GetConfigOrDie()
+
+	customCache, err := cache.New(configuration, cache.Options{Scheme: RuntimeScheme})
```

Comment:
I am confused. `sigs.k8s.io/controller-runtime/pkg/cache` is being used as the new (controller) manager?

Why not this?
```go
import (
    ctrl "sigs.k8s.io/controller-runtime"    
)
mgr := ctrl.NewManager(ctrl.GetConfigOrDie(), options)

cachedClient := mgr.GetClient()
uncachedClient := mgr.GetAPIReader()

```

The manager internally creates two clients, one with caching capabilities on list and get ops and the other, without cached list and get ops. You can use them in other modules

Reply:
Now I see.